src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -877,6 +895,67 @@ ExtractFromExpressionWalker(Node *node, List **qualifierList)
 
 
 /*
+ * ExtractOuterJoinQualifiersWalker walks over a FROM expression, and finds all
+ * explicit qualifiers in the outer join expression and returns these qualifiers.
+ *
+ * Note that we check if the qualifier node in join node is a list node. If it
+ * is not a list node which is the case for subqueries, then we run
+ * eval_const_expressions(), canonicalize_qual() and make_ands_implicit()
+ * on the qualifier node and get a list of flattened implicitly AND'ed qualifier
+ * list. Actually in the planer phase of PostgreSQL these functions also run on
+ * subqueries but differently from the outermost query, they are run on a copy
+ * of parse tree and changes do not get persisted as modifications to the original
+ * query tree.
+ */
+static bool
+ExtractOuterJoinQualifiersWalker(Node *node, List **qualifierList)
+{
+	bool walkerResult = false;
+	if (node == NULL)
+	{
+		return false;
+	}
+
+	/*
+	 * Get qualifier lists of the join node. Note that in the case of subqueries,
+	 * PostgreSQL can skip simplifying, flattening and making ANDs implicit.
+	 * If qualifiers node is not a list, then we run these preprocess routines
+	 * on qualifiers node.
+	 */
+	if (IsA(node, JoinExpr))
+	{
+		List *joinQualifierList = NIL;
+		JoinExpr *joinExpression = (JoinExpr *) node;
+		Node *joinQualifiersNode = joinExpression->quals;
+		JoinType joinType = joinExpression->jointype;
+
+		/* we extract qualifiers from only outer join clauses */
+		if (joinQualifiersNode != NULL && joinType != JOIN_INNER)
```

Comment:
Should this be something like ``&& joinType == JOIN_LEFT`` ? (to be consistent with the comment)